File: linear_search/java/LinearSearch.java
public class LinearSearch {

    /**
     * This method performs a linear search in an array to find a specific element.
     * 
     * @param array The array to search through.
     * @param key The element to find in the array.
     * @return The index of the element if found, otherwise -1.
     */
    public static int linearSearch(int[] array, int key) {
        for (int index = 0; index < array.length; index++) {
            if (array[index] == key) {
                return index;
            }
        }
        return -1; // key not found in the array
    }

    public static void main(String[] args) {
        int[] data = {10, 20, 80, 30, 60, 50, 110, 100, 130, 170};

        int key = 110;
        int result = linearSearch(data, key);

        if (result == -1) {
            System.out.println("Element not found in the array.");
        } else {
            System.out.println("Element found at index: " + result);
        }
    }
}

File: linear_search/go/linear_search.go
package main

import (
	"fmt"
)

// linearSearch function takes an array and a target value.
// It returns the index of the target value if found in the array.
// If the target is not found, it returns -1.
func linearSearch(data []int, target int) int {
	for index, value := range data {
		if value == target {
			return index // Return the index where the target is found
		}
	}
	return -1 // Return -1 if the target is not found
}

func main() {
	data := []int{3, 5, 7, 9, 1, 2, 4, 6, 8, 0} // Sample array
	target := 4                                // Target value to search for

	result := linearSearch(data, target)
	if result != -1 {
		fmt.Printf("Target %d found at index %d\n", target, result)
	} else {
		fmt.Println("Target not found")
	}
}
